networkx/utils/backends.py

```diff
@@ -766,82 +774,71 @@ def __call__(self, /, *args, backend=None, **kwargs):
 
         if not backends:
             # Fast path if no backends are installed
-            if backend is not None:
+            if backend is not None and backend != "networkx":
```

Comment:
@MridulS, here is the fix for the issue you called out today. And to reiterate: if no backends are installed (not counting networkx), the fast path of lines 776-779 is still taken.